osu.Game/Skinning/PoolableSkinnableSample.cs

```diff
@@ -83,6 +83,14 @@ private void updateSample()
 
             bool wasPlaying = Playing;
 
+            if (activeChannel != null)
+            {
+                // when switching away from previous samples, we don't want to call Stop() on them as it sounds better to let them play out.
+                // this may change in the future if we use PoolableSkinSample in more locations than gameplay.
+                // we *do* want to turn off looping, else we end up with an infinite looping sample running in the background.
+                activeChannel.Looping = false;
```

Comment:
This is suspicious.

The `sampleContainer.Clear()` line below should dispose the `DrawableSample`, which should dispose the `Sample` (because the ctor param `disposeSampleOnDisposal` is left as default), which should dispose all contained sample channels, which stops them from playing (https://github.com/ppy/osu-framework/blob/bc9cc38a2513e99b38b5ade98461537fd341cab0/osu.Framework/Audio/Sample/SampleChannel.cs#L29-L35).

What am I missing here?

Reply:
Looks like the `LegacySkinSample` I implemented doesn't implement `IDisposable`, which means the sample it's encapsulating doesn't actually get the disposal flow fired. Let me try fixing that....

Reply:
Works, have amended. Good catch.